### PR TITLE
LPS-163692 master

### PR DESCRIPTION
**Pull request by rolandpakai**
This is a forward port of the fix from 71x/73x because of code cohesion.
(Can not be tested based on the 71x/73x reproductions)

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java [hunk 1]
```diff
@@ -54,17 +56,27 @@ public String replaceAttachmentElements(
 
 		StringBuffer sb = new StringBuffer(content.length());
 
+		Map<String, FileEntry> fileEntries = new HashMap<>();
+
 		while (matcher.find()) {
-			FileEntry tempAttachmentFileEntry = _getFileEntry(matcher);
+			String match = matcher.group(0);
```
Comment (adolfopa) [inline modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java:62]: Can you rename `match` to `fileName`?

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java [hunk 2]
```diff
@@ -54,17 +56,27 @@ public String replaceAttachmentElements(
...
+
+			FileEntry attachmentFileEntry = fileEntries.get(match);
```
Comment (adolfopa) [inline modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java:64]: Can you rename `attachmentFileEntry` to simply `fileEntry`?

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java [hunk 3]
```diff
@@ -54,17 +56,27 @@ public String replaceAttachmentElements(
...
+
+			if (attachmentFileEntry == null) {
+				FileEntry tempAttachmentFileEntry = _getFileEntry(matcher);
+
+				attachmentFileEntry = saveTempFileUnsafeFunction.apply(
+					tempAttachmentFileEntry);
```
Comment (adolfopa) [inline modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java:70]: Can you inline the `tempAttachmentFileEntry` variable?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java (hunk 3 above)
Comment (rolandpakai) [reply modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java]: See SF commit

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java (hunk 2 above)
Comment (rolandpakai) [reply modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java]: See SF commit

File: modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java (hunk 1 above)
Comment (rolandpakai) [reply modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java]: See SF commit

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:foward

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Hanlf** [comment]
Hi @rolandpakai 
there is a `cannot find symbol` error about HTMLImageAttachmentElementHandler.java, should be related to this PR, see [Jenkins console,](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/893/consoleFull) could you please take a look, thanks!
```     [exec] > Task :apps:translation:translation-translator-google-cloud:deployConfigs NO-SOURCE
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java:77: error: cannot find symbol
     [exec] 						match, attachmentFileEntry)));
     [exec] 						^
     [exec]   symbol:   variable match
     [exec]   location: class com.liferay.upload.web.internal.HTMLImageAttachmentElementHandler
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/upload/upload-web/src/main/java/com/liferay/upload/web/internal/HTMLImageAttachmentElementHandler.java:77: error: cannot find symbol
     [exec] 						match, attachmentFileEntry)));
     [exec] 						       ^
     [exec]   symbol:   variable attachmentFileEntry
     [exec]   location: class com.liferay.upload.web.internal.HTMLImageAttachmentElementHandler
     [exec] 2 errors
     [exec] Note: Some input files use unchecked or unsafe operations.
     [exec] Note: Recompile with -Xlint:unchecked for details.```

**Comment by rolandpakai** [comment]
Hi @Hanlf 

i fixed the errors.

Thanks, Roland

**Comment by rolandpakai** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by rolandpakai** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:forward

**Pull request closed without merging**